src/Evolution/Systems/GrMhd/GhValenciaDivClean/Subcell/PrimitiveGhostData.cpp

```diff
@@ -0,0 +1,57 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/GrMhd/GhValenciaDivClean/Subcell/PrimitiveGhostData.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Evolution/DgSubcell/Projection.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace grmhd::GhValenciaDivClean::subcell {
+auto PrimitiveGhostDataOnSubcells::apply(
+    const Variables<hydro::grmhd_tags<DataVector>>& prims,
+    const tnsr::aa<DataVector, 3, Frame::Inertial>& spacetime_metric,
+    const tnsr::iaa<DataVector, 3, Frame::Inertial>& phi,
+    const tnsr::aa<DataVector, 3, Frame::Inertial>& pi)
+    -> Variables<tags_for_reconstruction> {
```

Comment:
trailing return type syntax allows you to access member type aliases of the class, like `tags_for_reconstruction` :) Doxygen also displays better for long return types because then it does `auto` instead of `tuples::TaggedTuple<Aaaaa, Bbbbb, Ccccc, Dddddd, Eeeee>`